### DISCUSSION TITLE
Does OTLP receiver support UDP

### DISCUSSION THREAD

**Discussion by vikrambe** [open-telemetry/opentelemetry-collector#6016] [General]
Use case : I want to use OtelCollector as agent, and would want to configure OTLP receiver to listen on UDP port to avoid buffering on the app side.
I tried using below OTLP receiver config and ran into this issue.  After going through the code i don't see any implementation for UDP in specific. Am i missing anything here ? Please suggest?

**ISSUE**
Error: cannot start pipelines: listen udp 127.0.0.1:4317: address localhost:4317: unexpected address type
2022/09/02 14:15:05 collector server run finished with error: cannot start pipelines: listen udp 127.0.0.1:4317: address localhost:4317: unexpected address type
Exiting.

**Config based on https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/config.md#:~:text=in%20RFC%204007.-,transport,-string **
```
  otlp:
    protocols:
      grpc:
        endpoint: 0.0.0.0:4317
        transport: "udp"
```
Above documentation seems to be misleading.

**Comment by vikrambe** [comment]
@jpkrohling @bogdandrutu  Can you please review and comment on this ?

**Comment by jpkrohling** [comment]
That document seems to be auto-generated. The Server Settings might indeed allow receivers to listen on UDP, but there's no UDP support for gRPC at all, and no support on the OTLP receiver to accept batches via UDP.

I believe we should:

1) change that doc to remove mentions to UDP (might be tricky, as it seems to be auto-generated)
2) add validation to the OTLP receiver (and likely others?) to prevent users from setting UDP

**Reply by vikrambe** [reply]
@jpkrohling Thanks for quick response. Appreciate it. 
Supporting UDP receiver for agents is definitely a value add, it helps avoid back pressure build-up on the application side. Jaeger supports Thrift over UDP between app and jaeger agent. Any reason why this was not considered in OTLP receiver. Please, comment?

**Comment by jpkrohling** [comment]
I don't think there is a specific reason, it just wasn't implemented. Are you up to opening a PR with a concrete proposal?

**Comment by bogdandrutu** [comment]
UDP does not work with gRPC, and seems to not work with HTTP, the two L7 layers that we use. But we do support unixsockets if you are on the same machine.

**Reply by cwegener** [reply]
> UDP does not work with gRPC, and seems to not work with HTTP

Is there any interest in supporting HTTP/3 for OTLPHTTP?
